app/assets/javascripts/components/articles/available_article.jsx

```diff
@@ -60,16 +60,27 @@ export const AvailableArticle = createReactClass({
     const ratingMobileClass = `${ratingClass} tablet-only`;
     const articleLink = <a onClick={this.stop} href={article.url} target="_blank" className="inline">{article.formatted_title}</a>;
 
-    let actionButton;
+    let actionCell;
     // Show 'Select' button to students
     if (this.props.current_user.role === 0) {
-      actionButton = (
-        <button className="button dark" onClick={this.onSelectHandler}>{I18n.t('assignments.select')}</button>
+      actionCell = (
+        <td className="table-action-cell">
+          <button className="button dark" onClick={this.onSelectHandler}>{I18n.t('assignments.select')}</button>
+        </td>
       );
-    // Show 'Remove' button to admins and facilitators
+    // Show both 'Select'and Remove' button to admins and facilitators
     } else if (this.props.current_user.admin || this.props.current_user.role > 0) {
-      actionButton = (
-        <button className="button dark" onClick={this.onRemoveHandler}>{I18n.t('assignments.remove')}</button>
+      actionCell = (
+        <td className="table-action-cell">
+          <button className="button dark" onClick={this.onRemoveHandler}>{I18n.t('assignments.remove')}</button>
+        </td>
+      );
+    } else if (this.props.current_user.isNonStudent && this.props.current_user.isStudent) {
+      actionCell = (
+        <td className="table-action-cell">
+          <button className="button dark" onClick={this.onSelectHandler}>{I18n.t('assignments.select')}</button>
+          <button className="button dark" onClick={this.onRemoveHandler}>{I18n.t('assignments.remove')}</button>
+        </td>
```

Comment:
The logic looks right, but this could be done more clearly and with less duplication if the two buttons were set independently of each other. Maybe just leave the `td` there but empty if neither button is present?